compiler/rustc_query_system/src/ich/impls_syntax.rs

```diff
@@ -75,7 +75,8 @@ impl<'a> HashStable<StableHashingContext<'a>> for SourceFile {
             ref normalized_pos,
         } = *self;
 
-        (name_hash as u64).hash_stable(hcx, hasher);
+        // FIXME: Is this `truncate` useful? Using the full hash seems better.
```

Comment:
@michaelwoerister who introduces this

Reply:
I'm working on this in https://github.com/rust-lang/rust/pull/110367
The rabbit-hole goes ever deeper...

Reply:
This question is handled in other PRs